### DISCUSSION TITLE
How to reset User password ?

### DISCUSSION THREAD

**Discussion by ds2k5** [wavelog/wavelog#413] [Q&A] [answered]
Hello,
how to reset Users password when the "Password Rest eMail" will not send.
How to do this on the MariaDB ?

**Comment by HB9HIL** [comment, marked as answer]
Hello @ds2k5 

Start the mysql command line tool (acting here as `root`, if you don't you have to specify the user `sudo -u username -p mysql`) 
```
mysql
```

Select your wavelog database (change the name accordingly)
```
USE wavelog;
```

Then use this command to set the password to `wavelog` (set the correct username!)
```
UPDATE users SET user_password = '$2y$10$X27rnGP4c.h8Ov5cvHR.xeFlYuEXJgHBDNuuxII2PMlLEoRuWVx.2' WHERE user_name = 'your username';
```

Now you can log normally with your username and the password `wavelog`.
Change your password immediately

**Comment by albertodrago** [comment]
HI All 
i have the same problem whith my admin account!
I performed the reset password account bu update below 

MariaDB [**WAVELOG_DB**]> **UPDATE users SET user_password = 'my_new_password' WHERE user_name = 'iu2plz';**

but I can't login because my user result "LOCKED".
The application report this warning "Your account is locked, due to too many failed login-attempts. Please reset your password."

I tried but querying the database I can't find the possibility to make a user unlock  :(
Thanks adavanced for support

Alberto DRAGO
IU2PLZ

**Reply by HB9HIL** [reply]
Hello @albertodrago 

To unlock your user use this command:

```sql
UPDATE users SET login_attempts = 0 WHERE user_name = 'your username';
```

To reset your password use the tutorial above. It's important that you can't use your new password directly here. You have to use the same command as above which includes the password `wavelog` as hash. 

```sql
UPDATE users SET user_password = '$2y$10$X27rnGP4c.h8Ov5cvHR.xeFlYuEXJgHBDNuuxII2PMlLEoRuWVx.2' WHERE user_name = 'your username';
```

Then you can login using the password `wavelog`. You have to change your password afterwards in the WebUI

**Reply by albertodrago** [reply]
Solved! 
many thanks for support !! 
i hope see you on air 

Regards 
Alberto

**Reply by ea5wa** [reply]
> Hello @albertodrago
> 
> To unlock your user use this command:
> 
> ```sql
> UPDATE users SET login_attempts = 0 WHERE user_name = 'your username';
> ```
> 
> To reset your password use the tutorial above. It's important that you can't use your new password directly here. You have to use the same command as above which includes the password `wavelog` as hash.
> 
> ```sql
> UPDATE users SET user_password = '$2y$10$X27rnGP4c.h8Ov5cvHR.xeFlYuEXJgHBDNuuxII2PMlLEoRuWVx.2' WHERE user_name = 'your username';
> ```
> 
> Then you can login using the password `wavelog`. You have to change your password afterwards in the WebUI

Where should I run this command if my wavelog is installed on a wamp server (windows)?

Thanks and 73

**Comment by kmanwar89** [comment]
Hi @HB9HIL  - was hoping for some guidance here. I was able to login last night, but can't this morning 😅.  Checking your commands above, where exactly are you running the commands? Neither my `wavelog-main` nor `wavelog-db` containers have the `mysql` command available within them.

Are your instructions intended for a native install, as opposed to a containerized install? Are you able to offer guidance for how to do the password reset for a containerized install instead? Thanks!

**Reply by HB9HIL** [reply]
Sure:

```bash
docker exec -it wavelog-db mariadb -u wavelog -p wavelog -e "UPDATE users SET user_password = '$2y$10$X27rnGP4c.h8Ov5cvHR.xeFlYuEXJgHBDNuuxII2PMlLEoRuWVx.2' WHERE user_name = 'your username';"
```

Then you can login using the password `wavelog`. You have to change your password afterwards in the WebUI

This will ask you for the database password. If you don't know anymore you can find the password in your docker-compose.yml

**Reply by kmanwar89** [reply]
Wow you responded quickly! As it turns out, I was experimenting in the background before I even saw your response and actually ended up at the same conclusion you shared, so thank you :)

What's interesting is I reset the password, then was greeted with the "account lockout" error. I resolved that with the steps you shared above, and then the reset password took effect. So the summary of what I did was:

1. Open a terminal into my `wavelog-db` docker container with the command `docker exec -it wavelog-db /bin/bash`
2. Within this terminal, I issued the command `mariadb --user=MY_MARIADB_USER --pasword=MY_MARIADB_PASSWORD wavelog` to enter the MariaDB CLI - it wasn't aliased to `mysql`, so that explains why those instructions didn't work initially for me.
3. I performed the password reset with the hashed `wavelog` password as instructed using the command: `UPDATE users SET user_password = '$2y$10$X27rnGP4c.h8Ov5cvHR.xeFlYuEXJgHBDNuuxII2PMlLEoRuWVx.2' WHERE user_name = 'MY_USERNAME';`. Successful output is:

```
Query OK, 1 row affected (0.007 sec)
Rows matched: 1  Changed: 1  Warnings: 0
```

4. I tried logging in, but was greeted with the account lockout error message ("Your account is locked due to too many failed login-attempts. Please reset your password.").  I resolved this by following the instructions to unlock the account with `UPDATE users SET login_attempts = 0 WHERE user_name = 'MY_USERNAME;`. Successful output is:

```
Query OK, 1 row affected (0.003 sec)
Rows matched: 1  Changed: 1  Warnings: 0
```

5. Finally, I logged into Wavelog using MY_USERNAME and the password `wavelog`, which I'm happy to report worked perfectly!

Had I seen your response sooner, I probably could've combined this into one step, so thank you for sharing that, and thank you for answering my query so quickly - 73!
